partial_msg.go

```diff
@@ -147,22 +155,82 @@ func (pmm *partialMessageManager) Start(ctx context.Context) (<-chan *PartiallyV
 			}
 		}
 	}()
+
+	// Use a dedicated goroutine for chain broadcast to avoid any delay in
+	// broadcasting chains as it can fundamentally affect progress across the system.
+	go func() {
+		ticker := time.NewTicker(pmm.rebroadcastInterval)
+		defer func() {
+			ticker.Stop()
+			log.Debugw("Partial message manager rebroadcast stopped.")
+		}()
+
+		var current *chainexchange.Message
+		for ctx.Err() == nil {
+			select {
+			case <-ctx.Done():
+				return
+			case t := <-ticker.C:
+				if current != nil {
+					current.Timestamp = roundDownToUnixTime(t, pmm.rebroadcastInterval)
+					if err := pmm.chainex.Broadcast(ctx, *current); err != nil {
+						log.Errorw("Failed to re-broadcast chain.", "instance", current.Instance, "chain", current.Chain, "error", err)
+					}
+				}
+			case pending, ok := <-pmm.pendingChainBroadcasts:
+				if !ok {
+					return
+				}
+				switch {
+				case current == nil, pending.Instance > current.Instance:
+					// Either there's no prior chain broadcast or a new instance has started.
+					// Broadcast immediately and reset the timer to tick from now onwards. This is to
+					// re-align the chain rebroadcast relative to instance start.
+					current = &pending
+					current.Timestamp = roundDownToUnixTime(time.Now(), pmm.rebroadcastInterval)
+					if err := pmm.chainex.Broadcast(ctx, *current); err != nil {
+						log.Errorw("Failed to immediately re-broadcast chain.", "instance", current.Instance, "chain", current.Chain, "error", err)
+					}
+					ticker.Reset(pmm.rebroadcastInterval)
```

Comment:
Should we set the ticker to the rounded down time + interval instead, to line it up?
I don't think it matters; it just came up while I was reviewing.

Reply:
Yup that makes sense. Thanks Kuba

Reply:
On second thought I think this makes sense for the first firing of ticker but after that we want to use the interval. Right?

This makes the code complicated enough such that it puts me off from doing it. With the default interval of 2 seconds (and most likely sub 10s in any case?) I think it's not worth it. 

Going to merge as is. Please let me know if you think otherwise and I will adjust in a follow up PR 🙏